sympy/ntheory/ec.py

```diff
@@ -0,0 +1,360 @@
+from sympy.abc import x, y
+from sympy.core.compatibility import as_int, is_sequence
+from sympy.core.numbers import oo
+from sympy.core.relational import Eq
+from sympy.polys.domains import FiniteField, QQ, RationalField
+from sympy.solvers.solvers import solve
+
+from .factor_ import divisors
+from .residue_ntheory import sqrt_mod
+
+
+class EllipticCurve():
+    """
+    Create the following Elliptic Curve over domain.
+
+    `y^{2} + a_{1} x y + a_{3} y = x^{3} + a_{2} x^{2} + a_{4} x + a_{6}`
+
+    The default domain is ``QQ``. If no coefficient ``a1``, ``a2``, ``a3``,
+    it create curve as following form.
+
+    `y^{2} = x^{3} + a_{4} x + a_{6}`
+
+    Examples
+    ========
+
+    References
+    ==========
+
+    [1] J. Silverman "A Friendly Introduction to Number Theory" Third Edition
+    [2] http://mathworld.wolfram.com/EllipticDiscriminant.html
+    [3] G. Hardy, E. Wright "An Introduction to the Theory of Numbers" Sixth Edition
+
+    """
+
+    def __init__(self, a4, a6, a1=0, a2=0, a3=0, domain=QQ):
```

Comment:
It is probably efficient to use non-Basic domains internally but it is not necessary to have them as input parameters. I would suggest using `modulus=0` instead. Then domain would be `QQ` if `modulus` is 0 and `FF(modulus)` otherwise.

Reply:
okay.